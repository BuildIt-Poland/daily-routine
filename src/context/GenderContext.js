import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

import { MALE } from '../constants/genders';

const GenderContext = createContext();
const { Provider } = GenderContext;

function GenderProvider({ children }) {
  const handleGenderChange = gender => setGender(gender);

  const [gender, setGender] = useState(MALE);

  return <Provider value={{ gender, handleGenderChange }}>{children}</Provider>;
}

GenderProvider.propTypes = {
  children: PropTypes.node
};

export { GenderContext, GenderProvider };
