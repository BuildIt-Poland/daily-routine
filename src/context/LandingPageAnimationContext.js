import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

const LandingPageAnimationContext = createContext();
const { Provider } = LandingPageAnimationContext;

function LandingPageAnimationProvider({ children, history }) {
  const REDIRECT_DELAY = 1000;

  function redirectWithDelay(url) {
    setTimeout(() => {
      history.push(url);
    }, REDIRECT_DELAY);
  }

  function animateAndRedirect(redirectUrl) {
    toggleAnimation(state => ({
      ...state,
      isAnimated: true
    }));
    redirectWithDelay(redirectUrl);
  }

  function stopAnimation() {
    toggleAnimation(state => ({
      ...state,
      isAnimated: false
    }));
  }

  const initialState = {
    isAnimated: false,
    animateAndRedirect,
    stopAnimation
  };

  const [animation, toggleAnimation] = useState(initialState);

  return <Provider value={animation}>{children}</Provider>;
}

LandingPageAnimationProvider.propTypes = {
  children: PropTypes.node,
  history: PropTypes.object
};

export { LandingPageAnimationContext, LandingPageAnimationProvider };
