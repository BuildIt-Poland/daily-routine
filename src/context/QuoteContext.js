import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

const QuoteContext = createContext();
const { Provider } = QuoteContext;

function QuoteProvider({ children }) {
  const [quote, setQuote] = useState(null);

  const handleQuoteChange = quote => setQuote(quote);

  return <Provider value={{ quote, handleQuoteChange }}>{children}</Provider>;
}

QuoteProvider.propTypes = {
  children: PropTypes.node
};

export { QuoteContext, QuoteProvider };
