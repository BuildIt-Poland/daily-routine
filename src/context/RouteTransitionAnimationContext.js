import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

import { history } from '../types';

const RouteTransitionAnimationContext = createContext();
const { Provider } = RouteTransitionAnimationContext;

function RouteTransitionAnimationProvider({ children, history }) {
  const REDIRECT_DELAY = 250;

  function redirectWithDelay(url) {
    setTimeout(() => {
      history.push(url);
    }, REDIRECT_DELAY);
  }

  function animateAndRedirect(redirectUrl) {
    setIsAnimating(true);
    redirectWithDelay(redirectUrl);
  }

  function stopAnimation() {
    setIsAnimating(false);
  }

  const [isAnimating, setIsAnimating] = useState(false);

  return <Provider value={{ isAnimating, animateAndRedirect, stopAnimation }}>{children}</Provider>;
}

RouteTransitionAnimationProvider.propTypes = {
  children: PropTypes.node,
  history: history.isRequired
};

export { RouteTransitionAnimationContext, RouteTransitionAnimationProvider };
