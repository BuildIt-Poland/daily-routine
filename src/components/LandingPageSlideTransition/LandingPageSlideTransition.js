import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { animated, useSpring } from 'react-spring';
import { colorDarkBlue, colorWhite, zIndexModalOverlay } from '../../styles/designTokens';
import { LandingPageAnimationContext } from '../../context/LandingPageAnimationContext';

function HomePageAnimation({ onFinished }) {
  const windowHeight = window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;

  const baseStyles = {
    left: '0',
    top: -windowHeight,
    width: '100%',
    height: windowHeight,
    position: 'fixed',
    zIndex: zIndexModalOverlay,
    display: 'block'
  };

  const firstLayerAnimationStyles = useSpring({
    from: { ...baseStyles, background: colorWhite },
    to: async next => {
      await next({ ...baseStyles, top: windowHeight });
    },
    config: { mass: 1, tension: 200, friction: 1, duration: 900 },
    delay: 0
  });

  const secondLayerAnimationStyles = useSpring({
    from: { ...baseStyles, background: colorDarkBlue, top: -windowHeight / 2, height: windowHeight / 3 },
    to: async next => {
      await next({ ...baseStyles, top: windowHeight });
    },
    config: { mass: 1, tension: 80, friction: 10, duration: 700 },
    delay: 1000,
    // onFinished needs to be called on the last animated element (secondLayer finishes last)
    onRest: onFinished
  });

  const thirdLayerAnimationStyles = useSpring({
    from: { ...baseStyles, background: colorWhite, top: -windowHeight / 4, height: windowHeight / 4 },
    to: async next => {
      await next({ ...baseStyles, top: windowHeight });
    },
    config: { mass: 1, tension: 120, friction: 40, duration: 800 },
    delay: 800
  });

  return (
    <>
      <animated.div style={firstLayerAnimationStyles} />
      <animated.div style={secondLayerAnimationStyles} />
      <animated.div style={thirdLayerAnimationStyles} />
    </>
  );
}

HomePageAnimation.propTypes = {
  onFinished: PropTypes.func.isRequired
};

export default function LandingPageSlideTransition() {
  const { isAnimating, stopAnimation } = useContext(LandingPageAnimationContext);
  return isAnimating && <HomePageAnimation onFinished={stopAnimation} />;
}
