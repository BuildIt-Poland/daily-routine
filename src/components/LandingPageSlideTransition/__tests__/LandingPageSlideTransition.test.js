import React from 'react';
import { create } from 'react-test-renderer';

import { LandingPageAnimationContext } from '../../../context/LandingPageAnimationContext';
import LandingPageSlideTransition from '../LandingPageSlideTransition';

describe('COMPONENT - LandingPageSlideTransition', () => {
  it('renders correctly when "isAnimating" is  "true"', () => {
    const animateAndRedirect = () => {};
    const stopAnimation = () => {};
    const isAnimating = true;
    const component = create(
      <LandingPageAnimationContext.Provider value={{ animateAndRedirect, isAnimating, stopAnimation }}>
        <LandingPageSlideTransition />
      </LandingPageAnimationContext.Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
