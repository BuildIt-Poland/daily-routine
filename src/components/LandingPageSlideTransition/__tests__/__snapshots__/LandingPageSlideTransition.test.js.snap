// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - LandingPageSlideTransition renders correctly when "isAnimating" is  "true" 1`] = `
Array [
  <div
    style={
      Object {
        "background": "rgba(255, 255, 255, 1)",
        "display": "block",
        "height": 768,
        "left": "0",
        "position": "fixed",
        "top": -768,
        "width": "100%",
        "zIndex": 8000,
      }
    }
  />,
  <div
    style={
      Object {
        "background": "rgba(21, 88, 161, 1)",
        "display": "block",
        "height": 256,
        "left": "0",
        "position": "fixed",
        "top": -384,
        "width": "100%",
        "zIndex": 8000,
      }
    }
  />,
  <div
    style={
      Object {
        "background": "rgba(255, 255, 255, 1)",
        "display": "block",
        "height": 192,
        "left": "0",
        "position": "fixed",
        "top": -192,
        "width": "100%",
        "zIndex": 8000,
      }
    }
  />,
]
`;
